crypto-core/src/main/java/com/palantir/crypto/io/DecryptingSeekableInput.java

```diff
@@ -96,11 +109,29 @@ public void seek(long pos) throws IOException {
         }
 
         long prevBlockOffset = prevBlock * blockSize;
-        Cipher cipher = seekableCipher.seek(prevBlockOffset);
+        CryptoCipher cipher = seekableCipher.seek(prevBlockOffset);
         delegate.seek(prevBlockOffset);
 
+        // TODO(srand) This is duplicating work already done in AesCtrCipher#seek as a result of the public
```

Comment:
Would this be adding a method to `SeekableCipher` that returns the IV of the block at which we're going to start decrypting? Then each implementation could do something like:

```
private byte[] currIv;
public byte[] getCurrentIv() { return currIv; }

// In the constructor:
currIv = initIv;

// And in the seek method, use the code that already exists to update the value of currIv
```

Does that make sense, or did you have something entirely different in mind?